maker/src/metrics.rs

```diff
@@ -58,12 +69,42 @@ pub fn init_meter() -> PrometheusExporter {
     opentelemetry_prometheus::exporter(controller).init()
 }
 
-pub fn collect(node: Arc<Node<InMemoryStore>>) {
+pub fn collect(node: Arc<Node<InMemoryStore>>, health: Health) {
     let cx = opentelemetry::Context::current();
 
     let channels = node.channel_manager.list_channels();
     channel_metrics(&cx, channels);
     node_metrics(&cx, node);
+    health_metrics(&cx, &health);
+}
+
+fn health_metrics(cx: &Context, health: &Health) {
+    update_health_metric(cx, health.get_coordinator_status(), &COORDINATOR_STATUS);
+    update_health_metric(cx, health.get_orderbook_status(), &ORDERBOOK_STATUS);
+}
+
+/// Updates the health metric given a service status
+fn update_health_metric(cx: &Context, service_status: ServiceStatus, gauge: &ObservableGauge<u64>) {
+    match service_status {
+        ServiceStatus::Unknown => {
+            gauge.observe(cx, 0, &[KeyValue::new("status", "unknown")]);
+        }
+        ServiceStatus::Online => {
+            gauge.observe(cx, 1, &[KeyValue::new("status", "online")]);
+        }
+        ServiceStatus::Offline => {
+            gauge.observe(cx, 0, &[KeyValue::new("status", "offline")]);
+        }
```

Comment:
🚨 I don't think this works like you want: 

Our grafana agent will poll this metric ever X seconds but it will not reset an old value it has seen before, which means, with this setup, we might have a service which is online and offline at the same time, i.e. 

at 10:00:00 service was offline: we create the metric: 
`coordinator_status{status=offline}=0`

at 11:00:00 the service is online, we create the metric
`coordinator_status{status=online}=1`

This means, at 11:00:00, in our grafana dashboard we will have both values set: 
```
coordinator_status{status=offline}=0
coordinator_status{status=online}=1
```

Because `coordinator_status{status=offline}` and `coordinator_status{status=online}` are two different metrics.

Instead, you need to use the same metric to hold different states, e.g. 

- `coordinator_status=0` = offline
- `coordinator_status=1` = online
- `coordinator_status=2` = unknown --> with this we could then indicate different error codes.

Reply:
indeed, I had a wrong mental model how attributes for metrics work. Thank you for explanation, I've fixed it now.


cool idea with mapping `2 == unknown` .